cabal-install/src/Distribution/Client/Init/Interactive/Command.hs

```diff
@@ -416,24 +416,18 @@ languagePrompt flags pkgType = getLanguage flags $ do
     let h2010   = "Haskell2010"
         h98     = "Haskell98"
         ghc2021 = "GHC2021 (requires at least GHC 9.2)"
-          
+
     l <- promptList ("Choose a language for your " ++ pkgType)
       [h2010, h98, ghc2021]
       (DefaultPrompt h2010)
       Nothing
       True
 
-    if
-      | l == h2010       -> return Haskell2010
-      | l == h98         -> return Haskell98
-      | l == ghc2021     -> return GHC2021
-      | all isAlphaNum l -> return $ UnknownLanguage l
-      | otherwise        -> do
-        putStrLn
-          $ "\nThe language must be alphanumeric. "
-          ++ "Please enter a different language."
-
-        languagePrompt flags pkgType
+    case l of
+      h2010   -> return Haskell2010
```

Comment:
That's actually wrong: I know because I fell in the same trap. Haskell is not Prolog, it will not magically unify variables. In other words, the version you have right now will introduce a new, shadowing binding `h2010` (and others).

Reply:
lol what a moron I am, thanks (hopefully -Wall would have catched it)

Reply:
I can testify the same thing, coming from Erlang and Prolog I fell in similar traps ;)